server-common/src/main/java/org/apache/kafka/server/common/TestFeatureVersion.java

```diff
@@ -16,22 +16,57 @@
  */
 package org.apache.kafka.server.common;
 
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.Map;
 
 public enum TestFeatureVersion implements FeatureVersion {
     TEST_0(0, MetadataVersion.MINIMUM_KRAFT_VERSION, Collections.emptyMap()),
     // TEST_1 released right before MV 3.7-IVO was released, and it has no dependencies
     TEST_1(1, MetadataVersion.IBP_3_7_IV0, Collections.emptyMap()),
-    // TEST_2 is not yet released and maps to the latest testing version, and it depends on this metadata version
-    TEST_2(2, MetadataVersion.latestTesting(), Collections.singletonMap(MetadataVersion.FEATURE_NAME, MetadataVersion.latestTesting().featureLevel()));
+    // TEST_2 is not yet set to be the default version and maps to the latest testing version, and it depends on this metadata version
+    TEST_2(2, MetadataVersion.latestTesting(), Collections.singletonMap(MetadataVersion.FEATURE_NAME, MetadataVersion.latestTesting().featureLevel())),
+
+    /**
+     * Test versions only used for unit test FeatureTest.java.
+     */
+    // For testing latest production is not one of the feature versions.
+    UT_FV0_0(0, MetadataVersion.MINIMUM_KRAFT_VERSION, Collections.emptyMap()),
```

Comment:
One issue with putting all unit test features here is that all of them will have the same feature name `test.feature.version`, which is confusing. Perhaps it's better to pull each unit test feature to its own file?